app/assets/javascripts/views/selectors_view.js

```diff
@@ -0,0 +1,40 @@
+(function(App) {
+
+  'use strict';
+
+  App.View = App.View || {};
+
+  App.View.Selectors = Backbone.View.extend({
+
+    events: {
+      'change' : 'onChangeTriggerValue'
+    },
+
+    initialize: function(settings) {
+      if (!this.el) {
+        return;
+      }
+
+      var opts = settings && settings.options ? settings.options : {};
+      this.options = _.extend({}, this.defaults, opts);
+
+      this.listeners();
+    },
+
+    listeners: function() {
+      Backbone.Events.on('map:country:selected', this.setSelectedValue.bind(this));
+    },
+
+    setSelectedValue: function(obj) {
+      this.$el.val(obj.name);
```

Comment:
The event handler that currently triggers the "fetch states" is set with jQuery on $("#country"). I'm guessing that by changing the selected value with `this.$el.val(value)` the change event is not triggered.
To get it to trigger you can call: `this.$el.trigger('change');`, probably before you can `this.$el.trigger('chosen:updated');` . This makes it work.